linkerd/app/inbound/src/lib.rs

```diff
@@ -266,21 +267,28 @@ where
                     .stack
                     .push_map_target(TcpEndpoint::from)
                     .push(self.runtime.metrics.transport.layer_accept())
-                    .push_map_target(TcpAccept::port_skipped)
-                    .instrument(|_: &_| debug_span!("forward"))
+                    .push_request_filter(TcpAccept::port_skipped)
+                    .check_new_service::<T, _>()
+                    .instrument(|_: &T| debug_span!("forward"))
                     .into_inner(),
             )
-            .check_new_service::<listen::Addrs, I>()
+            .check_new_service::<T, I>()
             .push_switch(
-                PreventLoop::from(server_port),
+                PreventLoop::from(server_port).to_switch(),
```

Comment:
Can we consolidate this switch to actually build a target type that holds the origdst? It looks like we have to handle this case with an error in at least 3 places... Not a blocker, though, as I understand followups may ease this...

I also find switches harder to understand with the impulse decoupled from the stack -- it'd probably be easier to follow by embedding the closure here directly...

Reply:
> Can we consolidate this switch to actually build a target type that holds the origdst? It looks like we have to handle this case with an error in at least 3 places... Not a blocker, though, as I understand followups may ease this...

the intent was to follow up with a branch that changes the whole stack to be based on `Param<OrigDstAddr>` rather than `Param<Option<OrigDstAddr>>`, so this is kind of an intermediate state; if you'd prefer to avoid that, we could try to land the whole change in one bigger branch when it's ready?